### PR TITLE
Added a BufferLogger command to open a file with a timestamp

### PR DESCRIPTION
**Pull request by kubiak-jpl**
Add a `BL_OpenFileTimestamp` command to BufferLoggers which will add a seconds and microseconds timestamp to data product filenames.

### CODE REVIEW

**Review by sivarticus — COMMENTED** [review]
Looks good. We might want to clear the nameTimestamp in the void BufferLogger::File::close(void) function. That way, the system would be reset to its default init configuration in case you wanted to do some non time-based logging

File: Svc/BufferLogger/BufferLoggerFile.cpp [hunk 1]
```diff
@@ -130,13 +148,25 @@ namespace Svc {
...
+        this->name.format(
+            "%s%s%05d%s",
+            this->prefix.toChar(),
+            this->baseName.toChar(),
+            this->fileCounter,
+            this->suffix.toChar()
+        );
+    } else {
+        this->name.format(
+            "%s%s%05d_%u_%06u%s",
```
Comment (sivarticus) [inline Svc/BufferLogger/BufferLoggerFile.cpp:161]: Good job not repeating the fprime gds usec bug here

**Comment by kubiak-jpl** [comment]
Adding it to the close function would prevent subsequent log files from including the timestamp. The timestamp is reset in both the `setBaseName` and `setBaseNameTimestamp` methods, which should cover all cases

**Comment by sivarticus** [comment]
> Adding it to the close function would prevent subsequent log files from including the timestamp. The timestamp is reset in both the `setBaseName` and `setBaseNameTimestamp` methods, which should cover all cases

Ahh, yeah, I missed that line in `setBaseName`. Looks good.

**Pull request merged**